scripts/aws.sh

```diff
@@ -0,0 +1,488 @@
+#!/bin/bash
+
+set -uo pipefail
+
+export AWS_PAGER=""
+
+err_log="err.log"
+
+trap 'rm -f "$err_log"' EXIT
+
+declare -A ami_map=(
+  [us-west-2,x86_64,rhel-9.2]=ami-0378fd0689802d015    # RHEL-9.2.0_HVM-20240229-x86_64-33-Hourly2-GP3
+  [us-west-2,x86_64,rhel-9.3]=ami-0c2f1f1137a85327e    # RHEL-9.3.0_HVM-20240229-x86_64-27-Hourly2-GP3
+  [us-west-2,x86_64,rhel-9.4]=ami-0423fca164888b941    # RHEL-9.4.0_HVM-20240605-x86_64-82-Hourly2-GP3
+  [us-west-2,arm64,rhel-9.2]=ami-0cb125bb261a63f52     # RHEL-9.2.0_HVM-20240229-arm64-33-Hourly2-GP3
+  [us-west-2,arm64,rhel-9.3]=ami-04379fa947a959c92     # RHEL-9.3.0_HVM-20240229-arm64-27-Hourly2-GP3
+  [us-west-2,arm64,rhel-9.4]=ami-05b40ce1c0e236ef2     # RHEL-9.4.0_HVM-20240605-arm64-82-Hourly2-GP3
+  [eu-west-1,x86_64,rhel-9.2]=ami-02c220fcee5dab581    # RHEL-9.2.0_HVM-20240229-x86_64-33-Hourly2-GP3
+  [eu-west-1,x86_64,rhel-9.3]=ami-05463a02d11667441    # RHEL-9.3.0_HVM-20240229-x86_64-27-Hourly2-GP3
+  [eu-west-1,x86_64,rhel-9.4]=ami-07d4917b6f95f5c2a    # RHEL-9.4.0_HVM-20240605-x86_64-82-Hourly2-GP3
+  [eu-west-1,arm64,rhel-9.2]=ami-06f9fb7baa169fdfc     # RHEL-9.2.0_HVM-20240229-arm64-33-Hourly2-GP3
+  [eu-west-1,arm64,rhel-9.3]=ami-016e6894567bb7f3e     # RHEL-9.3.0_HVM-20240229-arm64-27-Hourly2-GP3
+  [eu-west-1,arm64,rhel-9.4]=ami-02b8573b23fde21aa     # RHEL-9.4.0_HVM-20240605-arm64-82-Hourly2-GP3
+)
+
+spin() {
```

Comment:
This is nice for interactive use, but the `\r` would not work without a terminal. Practically, we can detect STDIN and only display spinner then.
However, let's think if we can just add "Started action" / "Finished action" messages?